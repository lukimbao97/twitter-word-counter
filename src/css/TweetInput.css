.btn-textarea-word-counter-container {
  margin-top: 35%;
}

.btn-wrapper {
  margin: 0 10px 0 auto;
  max-width: 10%;
}

.tweet-btn {
  border-radius: 50%;
  padding: 10px;
  background-color: rgb(29, 161, 242);
  cursor: pointer;
  color: white;
  outline-style: none;
  border-color: rgba(0, 0, 0, 0);
}

.twitters-textarea-and-img-container {
  display: flex;
}

.tweet-btn:disabled {
  opacity: 0.5;
}

textarea {
  margin: 10px;
  padding: 10px;
  width: 100%;
  border: none;
  font-family: "Inter", sans-serif;
  font-size: 20px;
}

textarea:focus {
  outline: none;
}
