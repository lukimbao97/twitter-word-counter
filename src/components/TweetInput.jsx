import axios from 'axios';
import React, { useState } from 'react';
import WordCounter from './WordCounter';
import "../css/TweetInput.css";

const URL = 'https://desolate-springs-52160.herokuapp.com';

const limitChara = 50;

export const handleCircleColor = (tweetValueLength) => {
    let color = "";

    if (limitChara - tweetValueLength < 0) color = "-red";
    else if (limitChara - tweetValueLength <= 10) color = "-yellow";

    return color;
};

export const handleCountWords = (tweetValueLength) => {
    return limitChara - tweetValueLength;
};

function TweetInput() {
    const [tweet, setTweet] = useState('');
    const [tweetObj] = useState({ userName: 'BaoLK', tagName: '@Bao', avatar: 'https://picsum.photos/id/237/50/50', tweet: '' })

    const compareChara = () => {
        if (tweet.length === 0) return true;
        return limitChara - tweet.length < 0 ? true : false;
    };

    const handlePostTweet = async () => {
        const copyTweetObj = { ...tweetObj };
        copyTweetObj.tweet = tweet;
        await axios.post(`${URL}/api/tweets`, copyTweetObj);
        setTweet('');
    }

    return (
        <div className="btn-textarea-word-counter-container">
            <div className="btn-wrapper">
                <button
                    className="tweet-btn"
                    onClick={handlePostTweet}
                    disabled={compareChara()}
                >
                    Tweet
                </button>
            </div>
            <div className="twitters-textarea-and-img-container">
                <img
                    src="https://picsum.photos/id/237/50/50"
                    alt=""
                    style={{ maxHeight: "50px", marginTop: "10px" }}
                />
                <textarea
                    cols="4"
                    rows="6"
                    value={tweet}
                    onChange={e => setTweet(e.target.value)}
                    placeholder="What's happening..."
                />
            </div>
            <WordCounter
                handleCircleColor={() => handleCircleColor(tweet.length)}
                handleCountWords={() => handleCountWords(tweet.length)}
            />
        </div>
    );
}

export default TweetInput;