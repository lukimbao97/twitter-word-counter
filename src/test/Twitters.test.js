import React from "react";
import renderer from "react-test-renderer";
import TweetInput, {
  handleCircleColor,
  handleCountWords,
} from "../components/Twitters";

// Test handleCircleColor function
test("test-handle-color-yellow", () => {
  const handleCircleColorResult = handleCircleColor(49);

  expect(handleCircleColorResult).toBe("-yellow");
});

test("test-handle-color-red", () => {
  const handleCircleColorResult = handleCircleColor(60);

  expect(handleCircleColorResult).toBe("-red");
});

test("test-handle-color", () => {
  const handleCircleColorResult = handleCircleColor(10);

  expect(handleCircleColorResult).toBe("");
});

// Test handleCountWords function
test("test-count-words-red", () => {
  const handleCountWordsResult = handleCountWords(65);

  expect(handleCountWordsResult).toBe(-15);
});

test("test-count-words-yellow", () => {
  const handleCountWordsResult = handleCountWords(45);

  expect(handleCountWordsResult).toBe(5);
});

test("test-count-words-normal", () => {
  const handleCountWordsResult = handleCountWords(20);

  expect(handleCountWordsResult).toBe(30);
});

//Test matchSnapShot for Twitters component
test("test-changes-from-Twitters-component", () => {
  const component = renderer.create(<TweetInput />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
