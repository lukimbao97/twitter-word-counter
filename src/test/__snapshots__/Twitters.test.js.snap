// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test-changes-from-Twitters-component 1`] = `
<div
  className="twitters-main-container"
>
  <div
    className="tweet-main-container"
  >
    <div>
      <img
        alt=""
        src="https://source.unsplash.com/random/50x50"
      />
    </div>
    <div
      className="tweet-container"
    >
      Just upgraded to latest chrome and experiencing this unwelcome ... Anyone know how to go back to normal?
    </div>
  </div>
  <div
    className="tweet-main-container"
  >
    <div>
      <img
        alt=""
        src="https://source.unsplash.com/random/50x50"
      />
    </div>
    <div
      className="tweet-container"
    >
      Just upgraded to latest chrome and experiencing this unwelcome ... Anyone know how to go back to normal?
    </div>
  </div>
  <textarea
    cols="4"
    onChange={[Function]}
    placeholder="What's happening..."
    rows="6"
    value=""
  />
  <div
    className="world-count-wrapper"
  >
    <span>
      50
    </span>
  </div>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Post
  </button>
</div>
`;
